instrumentation/servlet/src/main/java/brave/servlet/HttpServletHandler.java

```diff
@@ -0,0 +1,62 @@
+package brave.servlet;
+
+import brave.http.HttpAdapter;
+import brave.http.HttpServerHandler;
+import brave.http.HttpServerParser;
+import brave.internal.Nullable;
+import java.nio.ByteBuffer;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import zipkin.Endpoint;
+
+import static brave.internal.InetAddresses.ipStringToBytes;
+
+public final class HttpServletHandler // public for others like sparkjava to use
+    extends HttpServerHandler<HttpServletRequest, HttpServletResponse> {
+
+  public HttpServletHandler(HttpServerParser<HttpServletRequest, HttpServletResponse> parser) {
+    super(new HttpServletAdapter(), parser);
+  }
+
+  @Nullable public Endpoint.Builder remoteEndpoint(HttpServletRequest request) {
+    byte[] addressBytes = ipStringToBytes(request.getHeader("X-Forwarded-For"));
+    if (addressBytes == null) addressBytes = ipStringToBytes(request.getRemoteAddr());
```

Comment:
ex we are overriding here because servlet has a `getRemoteAddr` function which not all libraries have. The default we override looks at X-Forwarded-For only